### PR TITLE
fix(usb_host): Light sleep auto suspend hardening

### PR DESCRIPTION
**Pull request by peter-marcisovsky**
Fixing some smaller issues for the light sleep auto suspend feature:
 - fixing hub_root_can_resume allowed states
 - fixing stale hub state when device disconnects in light sleep
 - add limitations to the docs
 - add tests

---

## Checklist

Before submitting a Pull Request, please ensure the following:

- [ ] 🚨 This PR does not introduce breaking changes.
- [ ] All CI checks (GH Actions) pass.
- [ ] Documentation is updated as needed.
- [ ] Tests are updated or added as necessary.
- [ ] Code is well-commented, especially in complex areas.
- [ ] Git history is clean — commits are squashed to the minimum necessary.

---

> [!NOTE]
> **Medium Risk**
> Changes affect root-port suspend/resume and auto-resume-on-transfer in light-sleep scenarios; scope is limited to hub PM logic with new tests, but incorrect handling could affect USB connectivity after sleep.
> 
> **Overview**
> Hardens **USB Host** behavior around **light-sleep auto suspend**, especially when the device disconnects while the hub still thinks the root port is suspended.
> 
> **Hub driver (`hub.c`)** clears stale suspend state on disconnect during light sleep (resets root port to enabled, clears `light_sleep_auto_pm`), cross-checks HCD port state in `hub_root_is_suspended()`, and tightens `hub_root_can_resume()` so only truly suspended/resuming ports are allowed (disconnect/recovery returns `ESP_ERR_NOT_ALLOWED`). Resume handling also treats `ESP_ERR_INVALID_STATE` like a benign port state change.
> 
> **USB Host library** logs a warning when auto-resume before transfer submit fails.
> 
> **Docs** document that device reconnection during light sleep is not detected and is handled like suspend/resume.
> 
> **CDC-ACM test** asserts no stale devices remain immediately after waking from light sleep before disconnect/reconnect events.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit ef7609b7edad62a3cffa79302ee10579c7e32e4d. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Review by tore-espressif — COMMENTED** [review]
Overall LGTM.
I'd like to test some of these edge cases on hardware and then approve

File: docs/en/usb_host.rst [hunk 1]
```diff
@@ -54,6 +54,7 @@ Currently, the Host Library and the underlying Host Stack has the following limi
     - The External Hub Driver: Remote Wakeup feature is not supported (External Hubs are active, even if there are no devices inserted).
     - The External Hub Driver: Doesn't handle error cases (overcurrent handling, errors during initialization etc. are not implemented yet).
     - The External Hub Driver: No Interface selection. The Driver uses the first available Interface with Hub Class code (09h).
+    - Light sleep: USB Host with a USB Device connected does not detect device reconnection during light sleep. Treats it as a suspend/resume cycle.
```
Comment (tore-espressif) [inline docs/en/usb_host.rst:57]: This is a major limitation. Let's have a deeper look before we settle for this

Comment (tore-espressif) [reply docs/en/usb_host.rst:57]: @peter-marcisovsky Targets with DWC2 version 4.30a can detect the disconnect when system is in light sleep.

Since this is not a regression, IMO we can merge and release this feature as is now.

If we can't find a clean solution then I propose: 
after reattach, the device will be in Powered state, so it won't respond to any packets with address !=0. When the Host lib detects no response from the device, we can assume disconnection and then reset the port

**Review by tore-espressif — APPROVED** [review]

File: docs/en/usb_host.rst (hunk 1 above)
Comment (peter-marcisovsky) [reply docs/en/usb_host.rst:57]: Thanks for investigation. I will look into it.
